### PR TITLE
docs(cypress): create how-to-add-cypress-tests doc

### LINKED ISSUE #38853: docs: add cypress tests documentaton to contribution guides
**Is your feature request related to a problem? Please describe.**

Had trouble getting up to speed with [Cypress](https://docs.cypress.io/guides/overview/why-cypress.html#In-a-nutshell) since there's currently no documentation on for how it's implemented in the code base.

**Describe the solution you'd like**

I suggest we create a documentation page for Cypress that outlines how to:

- 1) run the tests locally and 
- 2) create end-to-end tests using Cypress whenever adding to the code base.

This section could potentially be under the ["Open a pull request"](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request) section of "Code Contribution Guides".

**Describe alternatives you've considered**

A a link to the docs already on [cypress.io](https://docs.cypress.io/guides/overview/why-cypress.html#In-a-nutshell) could be added to the [introduction](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=how-to-open-a-pull-request-pr), but in my opinion this would probably be insufficient.

**Additional context**

- the PR where I created a Cypress test: #38806 
- Cypress.io [docs](https://docs.cypress.io/guides/overview/why-cypress.html#In-a-nutshell)

_ps: I'd love to give a PR for this a stab if it's decided it's needed._

### PR DESCRIPTION
**Pull request by luqven**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38853

This pull request commits the following changes:

- creates `how-to-add-cypress-tests.md`
- references cypress doc in `how-to-open-a-pull-request.md`
- adds cypress doc to the sidebar

I was not able to get the docs up and running locally (Windows on WSL-2 Ubuntu) using the [working-on-docs-theme](https://contribute.freecodecamp.org/#/how-to-work-on-the-docs-theme) guide.

I did my best to keep things short and to the point. There are several issues that contributors can run into that I don't delve into, like needing to run `mongod` with the `sudo` command in WSL-2. I'm open to feedback as to whether or not to include more details.

### CODE REVIEW

File: docs/how-to-open-a-pull-request.md [hunk 1]
```diff
@@ -65,7 +65,9 @@ Some examples of good PRs titles would be:
 
 5. Indicate if you have tested on a local copy of the site or not.
 
-   This is very important when making changes that are not just edits to text content like documentation or a challenge description. Examples of changes that need local testing include JavaScript, CSS, or HTML which could change the functionality or layout of a page.
+   - This is very important when making changes that are not just edits to text content like documentation or a challenge description. Examples of changes that need local testing include JavaScript, CSS, or HTML which could change the functionality or layout of a page.
+
+   - These changes should be accompanied by corresponding [Cypress integration tests](/how-to-add-cypress-tests).
```
Comment (ojeytonwilliams) [inline docs/how-to-open-a-pull-request.md:70]:
```suggestion
   - If your PR affects the behaviour of a page it should be accompanied by corresponding [Cypress integration tests](/how-to-add-cypress-tests).
```
We don't (yet) test for changes in style, so it's probably better to leave this a little more relaxed.

File: docs/how-to-add-cypress-tests.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    ```console
+    mongod
+    ```
+
+  - On Windows, you must specify the full path to the `mongod` binary
+
+    ```console
+    "C:\Program Files\MongoDB\Server\3.6\bin\mongod"
+    ```
```
Comment (ojeytonwilliams) [inline docs/how-to-add-cypress-tests.md:35]: Typically, on Linux at least, `mongod` is going to be running as service and not need manually starting each time.

The main thing is we can reuse the [existing docs](https://contribute.freecodecamp.org/#/how-to-setup-freecodecamp-locally?id=step-3-start-mongodb-and-seed-the-database) here and just remind people that it needs to be running.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for the PR, it looks good.  Some (very minor) nitpicks:

File: docs/how-to-add-cypress-tests.md (hunk 2 above)
Comment (luqven) [reply docs/how-to-add-cypress-tests.md]: Thanks for the feedback! Address both concerns in latest commits.

**Review by ojeytonwilliams — APPROVED** [review]
Thanks for making those changes.
LGTM :+1:

File: docs/how-to-add-cypress-tests.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+- Cypress tests are in the `./cypress` directory.
+
+- Cypress tests for a curriculum module are in the corresponding curriculum directory, i.e. `cypress/integration/learn/responsive-web-design/basic-css/index.js`.
+
+### How to run tests
+
+**1. Ensure that MongoDB and client applications are running**
+
+  - [Start MongoDB and seed the database](/how-to-setup-freecodecamp-locally?id=step-3-start-mongodb-and-seed-the-database)
```
Comment (ojeytonwilliams) [inline docs/how-to-add-cypress-tests.md:19]:
```suggestion
  - [Start MongoDB and seed the database](/how-to-setup-freecodecamp-locally#step-3-start-mongodb-and-seed-the-database)
```
Sorry, I missed this the first time.  If you use anchor links, Docsify automatically converts them to id query params as needed.  At the moment it's fine either way, but `?id=` is an implementation detail and could break in future.

File: docs/how-to-add-cypress-tests.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  - [Start the freeCodeCamp client application and API server](/how-to-setup-freecodecamp-locally?id=step-4-start-the-freecodecamp-client-application-and-api-server)
```
Comment (ojeytonwilliams) [inline docs/how-to-add-cypress-tests.md:21]:
```suggestion
  - [Start the freeCodeCamp client application and API server](/how-to-setup-freecodecamp-locally#step-4-start-the-freecodecamp-client-application-and-api-server)
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: docs/how-to-add-cypress-tests.md (hunk 4 above)
Comment (luqven) [reply docs/how-to-add-cypress-tests.md]: Thanks, good catch! Committed both suggestions.

**Review by ojeytonwilliams — APPROVED** [review]
Awesome, thanks @luqven. LGTM :+1:

**Review by raisedadead — APPROVED** [review]

**Pull request merged**